harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -131,6 +132,8 @@ def __init__(
         precision: Union[Literal[32], Literal[16]] = 32,
         amp_backend: Union[Literal["native"], Literal["apex"]] = "native",
         amp_level: Literal["O0", "O1", "O2", "O3"] = "O2",
+        gradient_clip_val: Union[int, float, None] = None,
+        gradient_clip_algorithm: Union[Literal["norm"], Literal["value"]] = "norm",
```

Comment:
Asking users to pass these values here is a functional solution, meaning that users can use it and it will work.

But it is not a very "PytorchLightning-native" solution.  It would be really nice if we could somehow hook directly into the `configure_gradient_clipping` call directly on the LightningModule that the user sets up.

I don't think this would be terribly hard either, though from looking at LightningModule.clip_gradients, we'd have to makes sure to patch clip_gradients() to not try and read `self.trainer`.  I think it would be a very nice user experience though.

What do you think?